airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -350,6 +353,24 @@ def _dynamic_stream_configs(
                 if "type" not in dynamic_stream:
                     dynamic_stream["type"] = "DeclarativeStream"
 
+                # Ensure that each stream is created with a unique name
+                name = dynamic_stream.get("name")
+
+                if name in seen_dynamic_streams:
+                    error_message = f"Dynamic streams list contains a duplicate name: {name}. Please contact Airbyte Support."
+                    failure_type = FailureType.system_error
```

Comment:
nit: Should this be a system error? As we are getting streams from an endpoint, this, in theory, would be an issue with such an endpoint. Unless we handled wrongly the info from such an endpoint, maybe?

Reply:
This is already `system_error` 😅

Reply:
Oh yes, that was my point, is not system error an indicator for "our" system? Im just trying to distinguish if this should be also some config error.
Sorry for the previous misleading comment.